enhancements/ingress/set_eip_nlb_ingress.md

```diff
@@ -0,0 +1,359 @@
+    type: LoadBalancerService
+```
+
+Ingress Operator will create an NLB Ingress Controller called `test` with the service type load balancer having the service annotation `service.beta.kubernetes.io/aws-load-balancer-eip-allocations`
+populated with the value from the eip-allocations field.
+
+#### Cluster administrator adds eip-allocations field the existing ingress controller
+
+Ingress Operator will create an NLB Ingress Controller with the service type load balancer having the service annotation `service.beta.kubernetes.io/aws-load-balancer-eip-allocations`
+populated with the value from the eip-allocations field.
+
+#### Cluster administrator updates eip-allocations field the existing ingress controller
+
+Ingress Operator will update an NLB Ingress Controller with the service type load balancer having the service annotation `service.beta.kubernetes.io/aws-load-balancer-eip-allocations`
+populated with the value from the eip-allocations field.
+
+### API Extensions
+- The API for setting `eip-allocations` field in the installer-config will look as follows in the file `pkg/types/aws/platform.go`:
+
+```go
+// Platform stores all the global configuration that all machinesets
+// use.
+type Platform struct {
+...
+    // https://docs.aws.amazon.com/AmazonECS/latest/developerguide/load-balancer-types.html#nlb 
+    // +optional 
+    LBType configv1.AWSLBType `json:"lbType,omitempty"`
+
+    // networkLoadBalancerParameters holds configuration parameters for an AWS
+    // network load balancer. Present only if type is NLB.
+    //
+    // +optional
+    NetworkLoadBalancerParameters *configv1.AWSNetworkLoadBalancerParameters `json:"networkLoadBalancer,omitempty"`
+}
+
+```
+
+- The API for the `eip-allocations` field in the ingress cluster object to store the value of the
+  field `eip-allocations` from the installer which the ingress operator uses to create the ingress controller
+  will look as follows in the file `config/v1/types_ingress.go`:
+
+```go
+// AWSIngressSpec holds the desired state of the Ingress for Amazon Web Services infrastructure provider.
+// This only includes fields that can be modified in the cluster.
+// +union
+type AWSIngressSpec struct {
+...
+    // +unionDiscriminator
+    // +kubebuilder:validation:Enum:=NLB;Classic
+    // +kubebuilder:validation:Required
+    Type AWSLBType `json:"type,omitempty"`
+
+    // networkLoadBalancerParameters holds configuration parameters for an AWS
+    // network load balancer. Present only if type is NLB.
+    //
+    // +optional
+    NetworkLoadBalancerParameters *AWSNetworkLoadBalancerParameters `json:"networkLoadBalancer,omitempty"`
+}
+
+// AWSNetworkLoadBalancerParameters holds configuration parameters for an
+// AWS Network load balancer.
+type AWSNetworkLoadBalancerParameters struct {
+    EIPAllocations EIPAllocations `json:"eip-allocations,omitempty"`
```

Comment:
Better json name would be `eipAllocations`.  Using a `-` is very unconventional.